### PR TITLE
Pad images to size requirements in call API

### PR DESCRIPTION
**Pull request by RunDevelopment**
Changes:
- Add `get_padding` for `SizeRequirements`. This method will return the necessary to make a size satisfy the requirements.
- Automatically pad (and later unpad) input images in the call API. This means that user code no longer has to handle padding and can run models on images of any size.
- Some additional checks to verify call API requirements.

### CODE REVIEW

**Comment by RunDevelopment** [comment]
Getting padding right turned out to be more interesting than I thought. The issue is that `torch.nn.functional.pad(t, (0, pad_w, 0, pad_h), "reflect")` only works if `pad_w` and `pad_h` are less than the width and height of the image respectively. So we first have to pad as much as we can with `reflect` and then use something else to cover the rest (if anything). Surprisingly non-trivial.

**Review by joeyballentine — APPROVED** [review]
